gendiff/utils.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, Dict
+
+
+def get_keys_groups(
+    dict1: Dict[str, Any], dict2: Dict[str, Any]
+) -> Dict[str, set[Any]]:
+    keys1_set = set(dict1.keys())
+    keys2_set = set(dict2.keys())
+    keys_groups = {}
```

Comment:
[optional] instead of doing dict initialization here and keys assignments later, consider doing it like so:

```python
...
return {
    "union": union,
    "common": common,
    "missing": missing,
}
```